### PR TITLE
Align with standard EKS bootstrap

### PR DESCRIPTION
**Pull request by errm**
This PR makes a few changes:

* Functionally the files output by the bootstrap process are much more similar to those generated by the bootstrap script used in the official AMI https://github.com/awslabs/amazon-eks-ami/blob/master/files/bootstrap.sh
* Use packr https://github.com/gobuffalo/packr to embed template files into the final binary rather than doing this by hand....

Breaking change:

In line with https://github.com/awslabs/amazon-eks-ami the name of the binary for https://github.com/kubernetes-sigs/aws-iam-authenticator has changed from `/usr/local/bin/heptio-authenticator-aws` to `/usr/local/bin/aws-iam-authenticator` users will need to rename, or symlink the binary... I could have maintained something special to check what it available on the system ... but this seems over the top.... 

Nice Improvements:

* Uses a pod-infra-container-image (pause container) from an ecr repo so pods should start (slightly) faster.

### CODE REVIEW

File: pkg/system/templates/etc/systemd/system/kubelet.service [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+  --cni-bin-dir=/opt/cni/bin \
+  --cni-conf-dir=/etc/cni/net.d \
+  --container-runtime=docker \
+  --network-plugin=cni \
+  --cgroup-driver=cgroupfs \
+  --register-node=true \
+  --kubeconfig=/var/lib/kubelet/kubeconfig \
+  --feature-gates=RotateKubeletServerCertificate=true \
+  --anonymous-auth=false \
+  --client-ca-file=/etc/kubernetes/pki/ca.crt $KUBELET_ARGS $KUBELET_MAX_PODS $KUBELET_EXTRA_ARGS
```
Comment (mtpereira) [inline pkg/system/templates/etc/systemd/system/kubelet.service]: `$KUBELET_EXTRA_ARGS` doesn't seem to be populate. Is this for additional user-provided configs? Should we document it?

Comment (mtpereira) [reply pkg/system/templates/etc/systemd/system/kubelet.service]: Now that I re-read AWS's boostrap.sh, I think we're missing a `/etc/systemd/system/kubelet.service.d/30-kubelet-extra-args.conf` template file.

Comment (errm) [reply pkg/system/templates/etc/systemd/system/kubelet.service]: Yeah lets skip this for now ... until we provide some mechanism to provide the args to ekstrap ...

Comment (errm) [reply pkg/system/templates/etc/systemd/system/kubelet.service]: It doesn't hurt to leave this here actually ... it is quite possible for someone to add there own drop in if they need to pass some extra args here ... there is no harm in this variable being here if it is not set though ...

Comment (mtpereira) [reply pkg/system/templates/etc/systemd/system/kubelet.service]: Should we drop a small note about it on the README then?

**Pull request merged**